File: libs/eave-stdlib-py/src/eave/stdlib/headers.py
# Eave-specific request headers
EAVE_TEAM_ID_HEADER = "eave-team-id"
EAVE_ACCOUNT_ID_HEADER = "eave-account-id"
EAVE_SIGNATURE_HEADER = "eave-signature"
EAVE_SIG_TS_HEADER = "eave-sig-ts"
EAVE_ORIGIN_HEADER = "eave-origin"
EAVE_REQUEST_ID_HEADER = "eave-request-id"
EAVE_CRON_DISPATCH_KEY_HEADER = "eave-cron-dispatch-key"
EAVE_CLIENT_ID_HEADER = "eave-client-id"
EAVE_CLIENT_SECRET_HEADER = "eave-client-secret"  # noqa: S105

EAVE_DEV_BYPASS_HEADER = "X-Google-EAVEDEV"
"""
This header can be used to bypass certain checks in development, like payload signing.
It works because Google removes all "X-Google-*" headers on incoming requests, so if this header
is present, we can be reasonably sure that this is a development machine.
"""

# Request headers added by Google
GCP_CLOUD_TRACE_CONTEXT = "X-Cloud-Trace-Context"
GCP_GAE_REQUEST_LOG_ID = "X-Appengine-Request-Log-Id"
GCP_GAE_TASK_EXECUTION_COUNT = "X-AppEngine-TaskExecutionCount"

MIME_TYPE_JSON = "application/json"
MIME_TYPE_TEXT = "text/plain"
ENCODING_GZIP = "gzip"
ENCODING_DEFLATE = "deflate"

File: libs/eave-stdlib-ts/src/headers.ts
export const EAVE_TEAM_ID_HEADER = "eave-team-id";
export const EAVE_ACCOUNT_ID_HEADER = "eave-account-id";
export const EAVE_SIGNATURE_HEADER = "eave-signature";
export const EAVE_SIG_TS_HEADER = "eave-sig-ts";
export const EAVE_ORIGIN_HEADER = "eave-origin";
export const EAVE_REQUEST_ID_HEADER = "eave-request-id";

/*
This header can be used to bypass certain checks in development, like payload signing.
It works because Google removes all "X-Google-*" headers on incoming requests, so if this header
is present, we can be reasonably sure that this is a development machine.
*/
export const EAVE_DEV_BYPASS_HEADER = "x-google-eavedev";

// For express Response.locals[EAVE_CTX_KEY]
export const EAVE_CTX_KEY = "eave-ctx";

// Request headers added by Google
export const GCP_CLOUD_TRACE_CONTEXT = "x-cloud-trace-context";
export const GCP_GAE_REQUEST_LOG_ID = "x-appengine-request-log-id";
export const GCP_GAE_TASK_EXECUTION_COUNT = "x-appengine-taskexecutioncount";

export const MIME_TYPE_JSON = "application/json";
